src/Simplify.elm

```diff
@@ -846,26 +845,16 @@ type alias ModuleContext =
     { lookupTable : ModuleNameLookupTable
     , moduleName : ModuleName
     , exposedVariantTypes : Exposed
-    , imports :
-        Dict
-            ModuleName
-            { alias : Maybe ModuleName
-            , exposed : Exposed
-            }
-    , rangesToIgnore : List Range
-    , rightSidesOfPlusPlus : List Range
+    , moduleBindings : Set String
+    , localBindings : RangeDict (Set String)
+    , rangesToIgnore : RangeDict ()
+    , rightSidesOfPlusPlus : RangeDict ()
```

Comment:
Going from `List Range` to `RangeDict ()` is totally unrelated to tracking local bindings. It shouldn't hurt anyone either but if you want, I can revert it and maybe open in a separate PR

Reply:
I think the idea when I made them into lists is that they're always going to be quite small, so a list would be fine in practice. I think it might be slightly slower with a RangeDict, but I'm fine with using it, since it's not going to impact the performance much if they're tiny collections anyway.